api/app/responses/api.py

```diff
@@ -82,7 +83,41 @@ def post(self):
                 db.session.add(answer)
             db.session.commit()
 
-            return {'id': response.id, 'user_id': user_id}, 201  # 201 is 'CREATED' status code
+            try:
```

Comment:
Can we refactor this into another method (eg send_confirmation) and call it from the both POST and PUT, whenever response.is_submitted is TRUE (ie. we only send a confirmation mail when the user submits their response, not when they just save it for later).

Reply:
Aha, gotcha :)